src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1284,5 +1281,45 @@ public void ChainWithMaxReorgPlusExtraHeadersIsCalled_AnotherChainIsPresented_Co
                 }
             }
         }
+
+        /// <summary>
+        /// Issue 29 @ Peer presents at least two headers. Those headers will be connected the tree. 
+        /// Then we save the first such connected block to variable X and simulate block downloaded for BOTH block.
+        /// Disconnect of the peer is then flollowed, which removes its chain from the tree. 
+        /// Partial validation succeeded is then called on X.
+        /// </summary>
+        [Fact]
+        public void ChainedHeaderIsRemovedFromTheTree_PartialValidationSucceededCalled_NothingIsReturned()
+        {
+            // Chain header tree setup. Initial chain has 5 headers.
+            // Example: h1=h2=h3=h4=h5.
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder()
+                .WithInitialChain(initialChainSize)
+                .Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            // Extend chaintip with 2 headers, i.e. h1=h2=h3=h4=h5=h6=h7.
+            const int chainExtension = 2;
+            chainTip = ctx.ExtendAChain(chainExtension, chainTip);
+            
+            // Peer 1 presents these 2 headers.
+            const int peer1Id = 1;
+            List<BlockHeader> listOfHeaders = ctx.ChainedHeaderToList(chainTip, chainExtension);
+            cht.ConnectNewHeaders(peer1Id, listOfHeaders);
+
+            // Download both blocks.
+            ChainedHeader firstPresentedHeader = chainTip.Previous;
+            cht.BlockDataDownloaded(chainTip, chainTip.Block);
+            cht.BlockDataDownloaded(chainTip.Previous, firstPresentedHeader.Block);
+
+            // Disconnect peer 1.
+            cht.PeerDisconnected(peer1Id);
+
+            // Attempt to call PartialValidationSucceeded on a saved bock.
+            List<ChainedHeader> headersToValidate = cht.PartialValidationSucceeded(firstPresentedHeader, out bool reorgRequired);
```

Comment:
Soon to be merged the change that renames reorganisation required to full validation required, please Rename here. Also please assert that it is false.